src/Poseidon/RagnarokServer.pm

```diff
@@ -1002,24 +1005,24 @@ sub SendCharacterList
 sub SendMapLogin {
 	my ($self, $client, $msg, $index) = @_;
 
-	# '0073' => ['map_loaded','x4 a3',[qw(coords)]]
-	my $data;
-	if ( $config{server_type} =~ /^kRO/ ) { # kRO Zero
-		$data .= pack("C*", 0x83, 0x02) . $accountID; #accountID
-		$client->send($data);
-
-		$data = pack("C*", 0xDE, 0x0A, 0x46, 0x00, 0x00, 0x00); # idk '-'
-		$client->send($data);
-
-		$data = pack("C*", 0xEB, 0x02) . pack("V", getTickCount) . getCoordString($posX, $posY, 1) . pack("C*", 0x05, 0x05, 0x00, 0x00); # map loaded
-		$client->send($data);
+	$client->send(pack("v a4", 0x0283, $accountID));
 
+	# mapLogin packet
+	if ($self->{type}->{$config{server_type}}->{charListPacket} eq '0x0A18') {
+		# '0A18' => ['map_loaded', 'V a3 C2 v C', [qw(syncMapSync coords xSize ySize font sex)]], # 14
+		$client->send(pack("v", 0x02EB) . pack("V", getTickCount) . getCoordString($posX, $posY, 1) . pack("C*", 0x05, 0x05) .  pack("C*", 0x05, 0x05));
+	} elsif ($self->{type}->{$config{server_type}}->{charListPacket} eq '0x02EB') {
+		# '02EB' => ['map_loaded', 'V a3 a a v', [qw(syncMapSync coords xSize ySize font)]], # 13
+		$client->send(pack("v", 0x02EB) . pack("V", getTickCount) . getCoordString($posX, $posY, 1) . pack("C*", 0x05, 0x05) .  pack("C*", 0x05, 0x05));
 	} else {
-		if ( $config{server_type} !~ /^bRO/ ) { $data .= $accountID; } #<- This is Server Type Based !!
-		$data = pack("C*", 0x73, 0x00) . pack("V", getTickCount) . getCoordString($posX, $posY, 1) . pack("C*", 0x05, 0x05);
-		$client->send($data);
+		# '0073' => ['map_loaded','x4 a3',[qw(coords)]]
+		$client->send(pack("v", 0x0073) . pack("V", getTickCount) . getCoordString($posX, $posY, 1) . pack("C*", 0x05, 0x05);
 	}
 
+	if ( $config{server_type} =~ /^kRO/ ) { # kRO
+		$client->send(pack("C*", 0xDE, 0x0A, 0x46, 0x00, 0x00, 0x00)); # Overweight percent
```

Comment:
```c++
struct PACKET_ZC_OVERWEIGHT_PERCENT {
	int16 packetType;
	uint32 percent;
} __attribute__((packed));
```
need to be replaced in `Sakexe_0` and `ServerType0`
~'0ADE' => ['overweight_percent', 'v V', [qw(len percent)]],#TODO~
```perl
#'0ADE' => ['overweight_percent', 'V', [qw(percent)]] # 6 TODO
```
RagnarokServer.pm:
~$client->send(pack("C*", 0xDE, 0x0A, 0x46, 0x00, 0x00, 0x00)); # Overweight percent~
```perl
$client->send(pack("v", 0x0ADE) . pack("V", 0x00);
```

Reply:
fixed at https://github.com/OpenKore/openkore/pull/3468/commits/96e2fb6443ccade23155da5d35ba5d1cfc421bbf